utils/text_formatting.tsx

```diff
@@ -1,6 +1,8 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+//eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+//@ts-ignore
```

Comment:
Why are you disabling type checking here? We should create a followup to re-enable this if it's too much to fix for now.

Reply:
Ya the types for XRegExp are wrong. I tried reverting XRegExp, but it seems that it was the typechecker not noticing, not the types actually changing. 
Created a ticket: 
https://mattermost.atlassian.net/browse/MM-26347